File: linked_list_find/linked_list_find.js

class Node {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}
​
// Approach: Iterative -> While head is not null, check if head value is equal to target and return true if it is else set the current node to next. If while loop completes to null, return false.
const linkedListFind = (head, target) => {
  let current = head
  while(current != null) {
    if(current.val == target) return true
    current = current.next
  }
  return false
}
​
// Approach: Recursive -> Base case, if head is None we return false. Base case, if head val is eq to target we return true. Recursive step, return the function calling the head.next and passing the target.
// const linkedListFind = (head, target) => {
//   if (head === null) return false
//   if (head.val === target) return true
//   return linkedListFind(head.next, target)
// }
​
// const a = new Node("a");
// const b = new Node("b");
// const c = new Node("c");
// const d = new Node("d");
​
// a.next = b;
// b.next = c;
// c.next = d;

File: linked_list_find/linked_list_find.py

class Node:
  def __init__(self, val):
    self.val = val
    self.next = None
​
# Approach: Iterative -> While head is not null, check if head value is equal to target and return true if it is else set the current node to next. If while loop completes to null, return false.
def linked_list_find(head, target):
    current = head
    while current is not None:
       if current.val == target:
          return True
       current = current.next
    return False
​
# Approach: Recursive -> Base case, if head is None we return false. Base case, if head val is eq to target we return true. Recursive step, return the function calling the head.next and passing the target.
# def linked_list_find(head, target):
#     if head is None:
#        return False
#     if head.val == target:
#        return True
#     return linked_list_find(head.next, target)
​
# a = Node("a")
# b = Node("b")
# c = Node("c")
# d = Node("d")
​
# a.next = b
# b.next = c
# c.next = d
​
# # a -> b -> c -> d
​
# linked_list_find(a, "c") # True
​
# a = Node("a")
# b = Node("b")